kolibri/utils/cli.py

```diff
@@ -206,6 +212,77 @@ def plugin(plugin_name, **args):
 
     conf.save()
 
+webpack_process = None
+webpack_terminating = False
+
+
+def kill_webpack_process():
+    global webpack_process
+    global webpack_terminating
+
+    if webpack_process and webpack_process.returncode is not None:
+        return
+
+    logger.info('Closing webpack process')
+
+    webpack_terminating = True
+
+    webpack_process.terminate()
+
+
+def start_webpack():
+    global webpack_process
+    global webpack_terminating
+
+    logger.info('Starting webpack process')
+
+    webpack_process = subprocess.Popen(
+        'npm run watch',
+        shell=True,
+        stdin=subprocess.PIPE,
+        stdout=sys.stdout,
+        stderr=sys.stderr)
+
+    if webpack_process.poll() is not None:
+        raise SystemError('Webpack process failed to start')
+
+    logger.info('Webpack process on pid {0}'.format(webpack_process.pid))
+
+    webpack_process.wait()
+
+    if webpack_process.returncode != 0 and not webpack_terminating:
+        logger.error("Webpack process exited unexpectedly.")
+
+
+def start(foreground=False, debug=False, args=[]):
+    """
+    Start the kolibri server
+    :param foreground: Run in the foreground
+    :param debug: Run Django debug server, and run npm run watch to build frontend assets
+    """
+
+    if foreground:
+        logger.info("Running Kolibri server in foreground.")
+    else:
+        logger.error("Running Kolibri as a background process is not yet supported.")
+        sys.exit(1)
+
+    logger.info("Stand by while the server loads its data.")
+
+    if debug:
```

Comment:
Don't recall that? But in any case, I would have to disagree with both of us then :D

I think we should stick to the runserver command. For KA Lite, we missed out on a lot of using `runserver` because we went down the "invent our own dev server" path with cherrypy.

Btw. I've looked at how `runserver` works. The autoreloader is not configurable at all, and it does a lot of crazy multi-platform configurability stuff.. so I think the best path would be to overload the `runserver` command, calling an original unmodified `runserver` command, just like you're doing above! But not in the `start` function :)